src/core/execute.c

```diff
@@ -201,6 +202,60 @@ static const char *exec_context_tty_path(const ExecContext *context) {
         return "/dev/console";
 }
 
+static int exec_context_tty_size(const ExecContext *context, unsigned *ret_rows, unsigned *ret_cols) {
+        _cleanup_free_ char *rowskey = NULL, *rowsvalue = NULL, *colskey = NULL, *colsvalue = NULL;
+        unsigned rows = UINT_MAX, cols = UINT_MAX;
+        const char *tty;
+        int r;
+
+        assert(context);
+        assert(ret_rows);
+        assert(ret_cols);
+
+        rows = context->tty_rows;
+        cols = context->tty_cols;
+
+        tty = exec_context_tty_path(context);
+        if (!tty || (rows != UINT_MAX && cols != UINT_MAX)) {
+                *ret_rows = rows;
+                *ret_cols = cols;
+                return 0;
+        }
```

Comment:
The fallback logic is a bit inconsistent. In order of priority we use: `UINT_MAX`, kernel command line, `context->tty_{rows,cols}`. But if we encounter a failure, we go back to `UINT_MAX`. It doesn't matter greatly, because the failure is mostly theoretical, but it still doesn't look nice.

Reply:
The order of priority is: `context->tty_{rows, cols}`, kernel command line, `UINT_MAX`, no? Only if a value from the context is set to `UINT_MAX` (which equals unset) do we try to use the value from the kernel cmdline, which makes sense to me?

Reply:
In `setup_input()`, the fallback values are set to `UINT_MAX`. Then `(void) exec_context_tty_size(…)` is called, and, when the context does *not* set both rows and cols, it may e.g. fail on alloc or in `proc_cmdline_get_key_many()`. In that case, the fallback of `UINT_MAX` is used, and the context gets ignored completely.